### PR TITLE
Natspec issues

### PR DESCRIPTION
**Pull request by MathisGD**
Fixes https://github.com/cantinasec/review-morpho/issues/12

> - [MorphoChainlinkOracleV2.sol#L128](https://github.com/morpho-org/morpho-blue-oracles/blob/af1a0d305071bbdcb38f1b7b128ba6d0c9f31684/src/morpho-chainlink/MorphoChainlinkOracleV2.sol#L128): "Chainlink" should be replaced with "Chainlink-compliant"

Completely removed "chainlink"

> - [IMorphoChainlinkOracleV2.sol#L25](https://github.com/morpho-org/morpho-blue-oracles/blob/af1a0d305071bbdcb38f1b7b128ba6d0c9f31684/src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2.sol#L25): These feeds are "chainlink-interface-compatible" but they could be feeds that are not from Chainlink itself. Consider to better clarify it.

I simply removed chainlink, as it doesn't add a lot here.

> - [IMorphoChainlinkOracleV2Factory.sol#L28](https://github.com/morpho-org/morpho-blue-oracles/blob/af1a0d305071bbdcb38f1b7b128ba6d0c9f31684/src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol#L28): the natspec comment should be re-written to use the "vault" term as plural

Actually there was a bigger problem, that I fixed. I also remove "feeds should be in the correct order" because usually these things are implied

> - [IMorphoChainlinkOracleV2Factory.sol#L43](https://github.com/morpho-org/morpho-blue-oracles/blob/af1a0d305071bbdcb38f1b7b128ba6d0c9f31684/src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol#L43): This factory does not create MM vaults but a MorphoChainlinkOracle. Replace "MetaMorpho vault" with "Morpho Chainlink oracle"

removed

> - [MorphoChainlinkOracleV2Test.sol#L103](https://github.com/morpho-org/morpho-blue-oracles/blob/af1a0d305071bbdcb38f1b7b128ba6d0c9f31684/test/MorphoChainlinkOracleV2Test.sol#L103) + [MorphoChainlinkOracleV2Test.sol#L114](https://github.com/morpho-org/morpho-blue-oracles/blob/af1a0d305071bbdcb38f1b7b128ba6d0c9f31684/test/MorphoChainlinkOracleV2Test.sol#L114): be consistent across all the tests (if there's not a specific reason to do otherwise). For the vast majority of instances you have used 1e18 but here and in testSDaiUsdcOracle you are using 10 ** 18

fixed

### CODE REVIEW

File: src/morpho-chainlink/MorphoChainlinkOracleV2.sol [hunk 1]
```diff
@@ -125,7 +125,7 @@ contract MorphoChainlinkOracleV2 is IMorphoChainlinkOracleV2 {
         // = 1e36 * (pB1 * 1e(-dB1) * pB2) / (pQ1 * 1e(-dQ1) * pQ2)
 
         // Let fpB1, fpB2, fpQ1, fpQ2 be the feed precision of the respective prices pB1, pB2, pQ1, pQ2.
-        // Chainlink feeds return pB1 * 1e(fpB1), pB2 * 1e(fpB2), pQ1 * 1e(fpQ1) and pQ2 * 1e(fpQ2).
+        // Chainlink-compliant feeds return pB1 * 1e(fpB1), pB2 * 1e(fpB2), pQ1 * 1e(fpQ1) and pQ2 * 1e(fpQ2).
```
Comment (MathisGD) [inline src/morpho-chainlink/MorphoChainlinkOracleV2.sol:128]:
```suggestion
        // Feeds return pB1 * 1e(fpB1), pB2 * 1e(fpB2), pQ1 * 1e(fpQ1) and pQ2 * 1e(fpQ2).
```

**Review by MerlinEgalite — APPROVED** [review]

**Review by Jean-Grimal — APPROVED** [review]

File: src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol [hunk 2]
```diff
@@ -21,11 +21,10 @@ interface IMorphoChainlinkOracleV2Factory {
     /// @dev Here is the list of assumptions that guarantees the oracle behaves as expected:
     /// - Feeds are either Chainlink-compliant or the address zero.
     /// - Feeds have the same behavioral assumptions as Chainlink's.
-    /// - Feeds are set in the correct order.
```
Comment (QGarchery) [inline src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol:24]: Should we do the same thing for feeds ? Ie:
`The feeds, if set, are Chainlink compliant.`
or actually:
`The feeds, if set, are Chainlink-interface compliant.`

File: src/morpho-chainlink/MorphoChainlinkOracleV2.sol [hunk 3]
```diff
@@ -52,12 +52,10 @@ contract MorphoChainlinkOracleV2 is IMorphoChainlinkOracleV2 {
     /// @dev Here is the list of assumptions that guarantees the oracle behaves as expected:
     /// - Feeds are either Chainlink-compliant or the address zero.
     /// - Feeds have the same behavioral assumptions as Chainlink's.
-    /// - Feeds are set in the correct order.
+    /// - The vaults, if set, are ERC4626-compliant.
     /// - Decimals passed as argument are correct.
-    /// - The vaults' sample shares quoted as assets and the base feed prices don't overflow when multiplied.
-    /// - The quote feed prices don't overflow when multiplied.
-    /// - Vaults are either ERC4626-compliant or the address zero.
-    /// @dev The base asset is the collateral token and the quote asset is the loan token.
```
Comment (QGarchery) [inline src/morpho-chainlink/MorphoChainlinkOracleV2.sol:60]: I think we wanted to keep this last line. I remember a discussion saying that it was quite handy for integrators to have this information here

Comment (Rubilmax) [reply src/morpho-chainlink/MorphoChainlinkOracleV2.sol:60]: "is expected to be"

File: src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol [hunk 4]
```diff
@@ -22,6 +22,7 @@ interface IMorphoChainlinkOracleV2Factory {
     /// - Feeds are either Chainlink-compliant or the address zero.
     /// - Feeds have the same behavioral assumptions as Chainlink's.
     /// - The vaults, if set, are ERC4626-compliant.
+    /// - The feeds, if set, are Chainlink-interface-compliant.
```
Comment (QGarchery) [inline src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol:25]: We can remove line 22 now "Feeds are either Chainlink-compliant or the address zero."

Comment (MerlinEgalite) [reply src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol:25]: agree

Comment (MathisGD) [reply src/morpho-chainlink/interfaces/IMorphoChainlinkOracleV2Factory.sol:23]: i removed also 
"Feeds have the same behavioral assumptions as Chainlink's."

**Review by QGarchery — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Review by Jean-Grimal — APPROVED** [review]

**Pull request merged**